lib/internal/http2/core.js

```diff
@@ -1970,16 +1970,21 @@ class Http2Stream extends Duplex {
   _final(cb) {
     const handle = this[kHandle];
     if (this.pending) {
-      this.once('ready', () => this._final(cb));
+      this.once('ready', function() {
+        process.nextTick(() => this._final(cb));
+      });
     } else if (handle !== undefined) {
       debugStreamObj(this, '_final shutting down');
-      const req = new ShutdownWrap();
-      req.oncomplete = afterShutdown;
-      req.callback = cb;
-      req.handle = handle;
-      const err = handle.shutdown(req);
-      if (err === 1)  // synchronous finish
-        return afterShutdown.call(req, 0);
+      // TODO: Why does this need to be in nextTick?
```

Comment:
@addaleax Might you have some insight here?

Reply:
What happens when it isn’t? :sweat_smile:

Reply:
Oh! I forgot about that small detail:

```bash
Path: parallel/test-http2-client-socket-destroy
Mismatched noop function calls. Expected exactly 1, actual 0.
    at Object.mustCall (/Users/ronagy/GitHub/nxtedition/node/test/common/index.js:324:10)
    at Http2Server.<anonymous> (/Users/ronagy/GitHub/nxtedition/node/test/parallel/test-http2-client-socket-destroy.js:19:29)
    at Http2Server.<anonymous> (/Users/ronagy/GitHub/nxtedition/node/test/common/index.js:358:15)
    at Http2Server.emit (events.js:304:20)
    at ServerHttp2Session.sessionOnStream (internal/http2/core.js:2739:19)
    at ServerHttp2Session.emit (events.js:304:20)
    at emit (internal/http2/core.js:285:8)
    at processTicksAndRejections (internal/process/task_queues.js:87:22)
Command: out/Release/node --expose-internals /Users/ronagy/GitHub/nxtedition/node/test/parallel/test-http2-client-socket-destroy.js
```

Reply:
Seems like some form of timing issue.